### PR TITLE
Now handles fopen with wb mode and fwrite

### PR DESCRIPTION
**Pull request by mistasse**

### CODE REVIEW

File: src/org/mozartoz/truffle/nodes/builtins/OSBuiltins.java [hunk 1]
```diff
@@ -206,9 +207,15 @@ protected OzNode castMode(OzNode value) {
...
-			assert mode == "rb";
 			try {
-				return new FileInputStream(new File(fileName));
+				switch (mode) {
+				case "rb":
+					return new FileInputStream(new File(fileName));
+				case "wb":
+					return new FileOutputStream(new File(fileName));
+				default:
+					throw new OzException(this, OzException.newSystemError(ERROR_FACTORY.newInstance("os", "os", "fopen", 2, "Opening mode not implemented")));
```
Comment (eregon) [inline src/org/mozartoz/truffle/nodes/builtins/OSBuiltins.java]: A bit of a long line :wink: (I would extract the error record in a local variable)

File: src/org/mozartoz/truffle/nodes/builtins/OSBuiltins.java [hunk 2]
```diff
@@ -228,13 +235,19 @@ Object fread(Object fileNode, Object count, Object end, OzVar actualCount) {
...
 		@Specialization
-		Object fwrite(Object fileNode, Object data) {
-			return unimplemented();
+		Object fwrite(FileOutputStream file, byte[] data) {
+			try {
+				file.write(data);
+			} catch (IOException e) {
+				throw new RuntimeException(e);
+			}
+			return (long) data.length;
```
Comment (eregon) [inline src/org/mozartoz/truffle/nodes/builtins/OSBuiltins.java]: Make the return value (of the fwrite function) `long` instead so the cast can be implicit :)

**Comment by eregon** [comment]
Looks good, well done and welcome to mozart-graal :smiley:

**Comment by mistasse** [comment]
Thank you! Tried to correct it.

**Comment by eregon** [comment]
Thank you and merged! :+1:

**Pull request merged**